pyproject.toml

```diff
@@ -0,0 +1,157 @@
+[build-system]
+requires = ["poetry-core>=1.0.0,<2.0", "poetry-dynamic-versioning>=1.0.0,<2.0.0"]
+build-backend = "poetry_dynamic_versioning.backend"
+
+[tool.poetry]
+name = "imap-processing"
+# Gets updated dynamically by the poetry-dynamic-versioning plugin
+version = "0.0.0"
+description = "IMAP Science Operations Center Processing"
+authors = ["IMAP SDC Developers <imap-sdc@lists.lasp.colorado.edu>"]
+readme = "README.md"
+include = ["imap_processing/_version.py"]
+license = "MIT"
+keywords = ["IMAP", "SDC", "SOC", "Science Operations"]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: MIT License",
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Software Development",
+    "Topic :: Scientific/Engineering",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX",
+    "Operating System :: Unix",
+    "Operating System :: MacOS",
+]
+
+# Exclude tests from build
+exclude = ["imap_processing/tests"]
+
+[tool.poetry.dependencies]
+astropy-healpix = ">=1.0"
+cdflib = "^1.3.1"
+imap-data-access = ">=0.28.0"
+python = ">=3.10,<4"
+space_packet_parser = "^5.0.1"
+spiceypy = ">=6.0.0"
+xarray = '>=2024.10.0'
+numpy = "<=3"
+sammi-cdf = "^1.0"
+scipy = "^1.13"
+
+# Optional dependencies
+numpydoc = {version="^1.5.0", optional=true}
+openpyxl = {version=">=3.0.7", optional=true}
+pandas = {version=">=2.0.0", optional=true}
+pre-commit = {version="^3.3.3", optional=true}
+pydata-sphinx-theme = {version="*", optional=true}
+pytest = {version=">=6.2.5", optional=true}
+pytest-cov = {version="^4.0.0", optional=true}
+pytest-xdist = {version="^3.2", optional=true}
+ruff = {version="==0.2.1", optional=true}
+sphinx = {version="*", optional=true}
+sphinxcontrib-openapi = {version="^0.8.3", optional=true}
+mypy = {version="1.10.1", optional=true}
+requests = {version = "^2.32.3", optional = true}
+healpy = {version = "^1.18.0", optional = true}
+netcdf4 = {version ="^1.7.2",  optional = true}
+
+[tool.poetry.extras]
+dev = ["pre-commit", "ruff", "mypy"]
+doc = ["numpydoc", "pydata-sphinx-theme", "sphinx", "sphinxcontrib-openapi"]
+test = ["openpyxl", "pytest", "pytest-cov", "pytest-xdist", "requests", "netcdf4"]
+tools = ["openpyxl", "pandas"]
+map_visualization = ["healpy"]
```

Comment:
I suggest to remove anything we don't need here. This list should just include the minimum set of dependencies needed to run the software. Looks like your main code just depends on `imap-data-access`, `numpy`, `matplotlib`, `xarray`, and `cdflib`. You will probably also need a few others like `mypy`, `ruff`, etc. as well, but there are a bunch here that can be removed.